packages/flutter/lib/src/material/focusable.dart

```diff
@@ -0,0 +1,297 @@
+      key: GlobalKey(debugLabel: widget.debugLabel),
+    );
+    _hasFocus = node.hasFocus;
+    node.addListener(_handleFocusChanged);
+  }
+
+  @override
+  void dispose() {
+    node.dispose();
+    super.dispose();
+  }
+
+  void _handleFocusChanged() {
+    if (_hasFocus != node.hasFocus) {
+      setState(() {
+        _hasFocus = node.hasFocus;
+      });
+      if (widget.onFocusChange != null) {
+        widget.onFocusChange(node.hasFocus);
+      }
+    }
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final FocusableNode newParent = Focusable.of(context);
+    newParent.reparent(node);
+  }
+
+  Decoration _getDecoration(BuildContext context) {
+    if (!widget.showDecorations) {
+      return null;
+    }
+    if (_hasFocus) {
+      return widget.focusedDecoration ?? Theme.of(context).focusableTheme.focusedDecoration;
+    } else {
+      return widget.unfocusedDecoration ?? Theme.of(context).focusableTheme.unfocusedDecoration;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return RawKeyboardListener(
+      focusNode: node,
+      onKey: (RawKeyEvent event) {
+        if (widget.onKey != null) {
+          widget.onKey(node, event);
+        }
+      },
+      child: Container(
+        decoration: _getDecoration(context),
+        child: _FocusableMarker(
+          key: node.key,
+          node: node,
+          child: widget.child,
+        ),
+      ),
+    );
+  }
+}
+
+class _FocusableMarker extends InheritedWidget {
+  const _FocusableMarker({
+    Key key,
+    @required this.node,
+    Widget child,
+  })  : assert(node != null),
+        super(key: key, child: child);
+
+  final FocusableNode node;
+
+  @override
+  bool updateShouldNotify(_FocusableMarker oldWidget) {
+    return node != oldWidget.node;
+  }
+}
+
+/// Defines the appearance of [Focusable]s.
+///
+/// This class is used to define the value of [ThemeData.focusableTheme]. The
+/// [Focusable] widget uses the current focusable theme to initialize
+/// some null [Focusable] properties (like [Focusable.focusedDecoration]).
+class FocusableThemeData extends Diagnosticable {
+  /// Creates a const FocusableThemeData object.
+  ///
+  /// All of the arguments are optional, and will use fallback values if not
+  /// specified.
+  const FocusableThemeData({Decoration focusedDecoration, Decoration unfocusedDecoration})
+      : focusedDecoration = focusedDecoration ?? _defaultFocusedDecoration,
+        unfocusedDecoration = unfocusedDecoration ?? _defaultUnfocusedDecoration;
+
+  static const Color _defaultFocusedColor = Color(0xff000080);
+  static const Color _defaultUnfocusedColor = Color(0x00000000);
+  static const double _defaultBorderWidth = 1.0;
+  static const BorderSide _defaultFocusedBorder = BorderSide(color: _defaultFocusedColor, width: _defaultBorderWidth);
+  static const BorderSide _defaultUnfocusedBorder = BorderSide(color: _defaultUnfocusedColor, width: _defaultBorderWidth);
```

Comment:
FWIW, not sure if FocusableThemeData belongs in material or widgets, but specific colors and thicknesses definitely shouldn't be in the widgets layer. Widgets layer should not have any opinion about drawing any pixels.

Reply:
Yeah, makes sense. When Focusable needs to get the theme data, it needs access to Theme.of, which is in material, so it has to be there, which made putting it here make sense.